miqa/core/models/group.py

```diff
@@ -0,0 +1,17 @@
+from uuid import uuid4
+from django.db import models
+
+
+class Group(models.Model):
```

Comment:
@zachmullen - I had initially named this `ArtifactGroup` but realized there were multiple items that could use similar grouping. Rather than creating a separate model for each type of item I created the generic `Group`. 

Other objects I thought could be similarly grouped (/made dynamically editable) in the future include:
- evaluation models (core\project.py, learning/evaluation_models.py)
- scan types (core\scan.py)
- decision choices (core\scan_decision.py)
- columns (learning\nn_training.py)

If you don't see any of these as likely candidates to become dynamic in the future / would rather have a separate model for each, I can definitely refactor back to `ArtifactGroup`, just lmk.

Reply:
Having those things dynamically editable sounds appealing to me.

Reply:
@zachmullen , are you recommending that we shouldn't use the Group model and instead be specific to only artifacts, then create other groups for other changeable features?  I like that a project can have a particular Group name, which can specify Artifcacts now and later specify the other options Dave mentioned (scans, learning columns, etc.)  Maybe I am just confused, but I liked the design as originally submitted.  Maybe I just need a bit of explanation.  thanks.

Reply:
Yes, that is what I was proposing. If we actually want multiple different types of things to be grouped together, then perhaps a generic `Group` is a good model, but that didn't seem like the design intent, as the field on the Project model was named `artifact_group`. I assumed we would only ever want to group things together of the same domain type.

Reply:
@zachmullen , Thanks.  It is my opinion that we would want to use the Group model to select sets of different things to go with projects.  In other words, the same way we have an Artifact model now with a group attribute, so we can build lists of Artifcats for each project , I think we might want to have the name of the model and its Evaluation columns (the outputs of the learning model) done similarly.    

I don't want to embark on this approach immediately across the board, but I'd like to leave room for generalizing the application of MIQA to a learning model designed for my Lung Cancer application, for example.

Reply:
If we do want multiple types of domain objects to be put into the same group, I'd still suggest we change the name to e.g.  `ProjectOptionsGroup`, to avoid collision with the existing `Group` model from django-contrib.

Reply:
@zachmullen - Definitely see how the `artifact_group` on Project could be misleading. If I understand correctly Django can support multiple fields to the same model with different names (e.g. `scan_group` also pointing to the `Group` model).

Changing the name from `Group` makes sense to me. Wondering if we should make it `ProjectOptionsGroup` or `OptionsGroup` to allow it to be used more generically for settings that could occur at other levels (e.g. global or perhaps per experiment)? I don't have a strong opinion on this, just wanted to raise the Q.

Reply:
Group has been recreated as SettingsGroup (meant it to be OptionsGroup but if it is all the same will leave as is, otherwise, easy change to OptionsGroup).